testing/upgradetest/versions.go

```diff
@@ -248,42 +246,73 @@ func findRequiredVersions(sortedParsedVersions []*version.ParsedSemVer, reqs Ver
 	return upgradableVersions, nil
 }
 
-// PreviousMinor returns the previous minor version available for upgrade.
 func PreviousMinor() (*version.ParsedSemVer, error) {
-	versions, err := GetUpgradableVersions()
+	currentVersion := define.Version()
+	upgradeableVersions, err := GetUpgradableVersions()
```

Comment:
Maybe we can keep it simpler? (cannot do a single suggestion block because of deleted lines but this should render the idea)
```suggestion
func PreviousMinor() (*version.ParsedSemVer, error) {
	return PreviousMinorVersion(false)
}

func PreviousMinorVersion(includeSnapshots bool) (*version.ParsedSemVer, error) {
	currentVersion := define.Version()
	upgradeableVersions, err := GetUpgradableVersions()
	if err != nil {
		return nil, err
	}

	return previousMinor(currentVersion, upgradeableVersions, includeSnapshots)
}
```